src/Avalonia.Animation/AnimatorStateMachine`1.cs

```diff
@@ -243,7 +243,10 @@ private void InternalStep(PlayState _playState, Func<double, T, T> Interpolator)
                         {
                             _targetControl.SetValue(_parent.Property, _lastInterpValue, BindingPriority.LocalValue);
                         }
+
                         _targetObserver.OnCompleted();
+                        _targetAnimation.SetDone(_targetControl);
```

Comment:
I've been doing some tests by re-implementing the page transitions, and I found a bug here. If the user animates a render transformation, then the value of `_targetControl` is the Transform object, not the control that is being transformed. This causes the animation to never be registered as completed.